lib/atlassian-jwt-authentication/filters.rb

```diff
@@ -24,16 +24,16 @@ def on_add_on_installed
       base_url = params[:baseUrl]
       api_base_url = params[:baseApiUrl] || base_url
 
+      # All install(including upgrades)/uninstall hooks will be asymmetrically
+      # signed with RS256 (RSA Signature with SHA-256) algorithm
+      return false unless _verify_jwt(addon_key, force_asymmetric_verify: true)
+
       jwt_auth = JwtToken.where(client_key: client_key, addon_key: addon_key).first
-      if jwt_auth
-        # The add-on was previously installed on this client
-        return false unless _verify_jwt(addon_key)
-        if jwt_auth.id != current_jwt_token.id
-          # Update request was issued to another plugin
-          render_forbidden
-          return false
-        end
-      else
+      if jwt_auth && jwt_auth.id != current_jwt_token.id
+        # Update request was issued to another plugin
+        render_forbidden
+        return false
+      elsif jwt_auth.nil?
```

Comment:
Inverting the if and elsif would make it easier to read imho:
```suggestion
      if jwt_auth.nil?
        self.current_jwt_token = JwtToken.new(jwt_token_params)
      elsif jwt_auth.id != current_jwt_token.id
        # Update request was issued to another plugin
        render_forbidden
        return false
```

Reply:
You're right, I updated it.